e2etest/bgptests/validate.go

```diff
@@ -152,3 +152,37 @@ func getIngressIPs(ingresses []corev1.LoadBalancerIngress) []string {
 	}
 	return ips
 }
+
+func validateServiceNotAdvertised(svc *corev1.Service, c *frrcontainer.FRR, ipFamily ipfamily.Family) {
+	for _, ip := range svc.Status.LoadBalancer.Ingress {
+		ingressIP := e2eservice.GetIngressPoint(&ip)
+
+		Eventually(func() bool {
+			frrRoutesV4, frrRoutesV6, err := frr.Routes(c)
+			if err != nil {
+				framework.ExpectNoError(err)
+			}
+
+			_, ok := frrRoutesV4[ingressIP]
+			if ipFamily == ipfamily.IPv6 {
+				_, ok = frrRoutesV6[ingressIP]
+			}
+
+			return ok
+		}, 4*time.Minute, 1*time.Second).Should(Equal(false))
+	}
+}
+
+func validateServiceInRoutesForCommunity(c *frrcontainer.FRR, community string, family ipfamily.Family, svc *corev1.Service) error {
```

Comment:
Put the eventually inside here, it will make the test body lighter

Reply:
i use this function twice, once the eventually expects no error, the other time it expects an error.